test/StreamableBodyTest.cpp

```diff
@@ -10,6 +10,33 @@
 using namespace ad_utility::httpUtils::httpStreams;
 using ad_utility::stream_generator::stream_generator;
 
+std::string_view toStringView(
+    const streamable_body::writer::const_buffers_type& buffer) {
+  return {static_cast<const char*>(buffer.data()), buffer.size()};
+}
+
+namespace boost {
+// For some reason those functions are required for the ASSERT_EQ/ASSERT_NE
+// macros to compile with boost::optional
+std::ostream& operator<<(std::ostream& out, none_t) {
+  out << "boost::none";
+  return out;
+}
+std::ostream& operator<<(
+    std::ostream& out,
+    const optional<std::pair<asio::const_buffer, bool>>& optionalBuffer) {

```

Comment:
Does Gtest provide `operator<<` for `std::pair`? Then it would suffice to provide this operator for the `asio:const_buffer`

Reply:
Good question. boost doesn't provide `operator<<` for `boost::optional` by default (you need to include an "optional_io" header (or similar, not sure how it's called), so that's why I chose to implement this in this rather specific way

Reply:
And no, `std::pair` doesn't seem to be printable, so I'll leave the code as-is